### ISSUE TITLE
NumberFormatException on Infinite Sorted Set Score

### ISSUE THREAD

**Issue by efmr** [xetorthio/jedis#1123]
Redis supports Infinite values on Sorted Sets by setting them to "inf" or "-inf". This is an issue for Jedis since any method that retrieves the score from the sorted set will throw an NumberFormatException since `Double.valueOf("inf")` is not a valid input string for conversion.

I suggest matching `Double.POSITIVE_INFINITY` (and negative) when the input string matches "inf" or"-inf".

Is there a workaround this issue?

Following is an example code that generates this exception.

``` java
import org.junit.Assert;
import org.junit.Test;
import redis.clients.jedis.Jedis;
import redis.clients.jedis.ZParams;

public class JedisResponseTest {
    @Test
    public void infiniteSortedSetScore() {
        Jedis jedis = new Jedis("localhost");
        /*
        Setting up a sorted set member score to be equal to inf since
          .zadd("foo", Double.POSITIVE_INFINITY, "bar");
        OR
          .zadd("foo", "inf", "bar");
        are not implemented
        */
        jedis.zadd("foo", Double.MAX_VALUE, "bar");
        ZParams params = new ZParams();
        params.weightsByDouble(2d);
        jedis.zinterstore("foobar", params, "foo");
        //should convert redis response "inf" to Double.POSITIVE_INFINITY
        // but throws numberFormatException instead
        Double result = jedis.zscore("foobar", "bar");
        Assert.assertTrue(result != null && result.equals(Double.POSITIVE_INFINITY));
        /*
        java.lang.NumberFormatException: For input string: "inf"
            at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1250)
            at java.lang.Double.valueOf(Double.java:504)
            at java.lang.Double.<init>(Double.java:597)
            at redis.clients.jedis.Jedis.zscore(Jedis.java:1474)
            at JedisResponseTest.infiniteSortedSetScore(JedisResponseTest.java:22)
        (...)
        */
    }
}
```

**Comment by HeartSaVioR** [comment]
Maybe we need to upmerge #463 and merge.

**Comment by sazzad16** [comment]
Resolved by #1642
